### PR TITLE
added container network metrics

### PR DESCRIPTION
**Pull request by JVecsei1**
* Added network metrics `RxBytes`/`TxBytes` [as discussed in here](https://cloudfoundry.slack.com/archives/C01ABMVNE9E/p1687523982698929?thread_ts=1683732484.532129&cid=C01ABMVNE9E).
* Related to https://github.com/cloudfoundry/cf-networking-release/issues/64 
* Depends on
  * https://github.com/cloudfoundry/cf-networking-release/pull/229
  * **Must be merged first:** https://github.com/cloudfoundry/garden/pull/112

### CODE REVIEW

File: gardener/gardener_test.go [hunk 1]
```diff
@@ -1708,9 +1723,19 @@ var _ = Describe("Gardener", func() {
...
+
+			containerizer.RunCalls(func(logger lager.Logger, s string, processSpec garden.ProcessSpec, io garden.ProcessIO) (garden.Process, error) {
+				_, _ = io.Stdout.Write([]byte(fmt.Sprintf("%d\n%d", networkStat.RxBytes, networkStat.TxBytes)))
+				return networkStatProcess, nil
+			})
+
 		})
 
-		It("should return the cpu and memory metrics the containerizer", func() {
+		It("should return the cpu and memory metrics from the containerizer", func() {
```
Comment (JVecsei1) [inline gardener/gardener_test.go:1738]: Unrelated

Comment (JVecsei1) [inline gardener/gardener_test.go:1738]: Unrelated cleanup, matching:

https://github.com/cloudfoundry/guardian/blob/01df65f1854af0cdaba941947984c72808ff256c/gardener/gardener_test.go#L1746

**Comment by geofffranks** [comment]
Code looks good. I added a question to [the originating issue](https://github.com/cloudfoundry/cf-networking-release/issues/64) regarding cross-compatibility with other releases (diego + cf-networking) that we'll want to ensure is tested prior to merging.

File: gardener/container.go [hunk 2]
```diff
@@ -145,16 +149,84 @@ func (c *container) Metrics() (garden.Metrics, error) {
...
+	}
+
+	return garden.ContainerNetworkStat{
+		RxBytes: rxBytes,
+		TxBytes: txBytes,
+	}, nil
+}
+
+func networkStatPath(ifName, stat string) string {
+	return filepath.Join("/sys/class/net", ifName, "statistics", stat)
```
Comment (geofffranks) [inline gardener/container.go:227]: This will likely fail on windows containers. We should make sure to test this as well.

Comment (JVecsei1) [reply gardener/container.go]: Hi @geofffranks,

we now added a noop implementation that is used on windows and in case the feature flag, [introduced here](https://github.com/cloudfoundry/garden-runc-release/pull/294), is disabled.

e87a8d4bb577828795ea3379718939a98b32c175

**Comment by geofffranks** [comment]
@JVecsei1 while testing i noticed there was an additional commit on the `add-network-metrics-v2` branch that didn't make it on this PR. is that desired here?

**Comment by JVecsei1** [comment]
> @JVecsei1 while testing i noticed there was an additional commit on the `add-network-metrics-v2` branch that didn't make it on this PR. is that desired here?

That's fine. I squashed two commits after adding them to the PR branch. `git diff add-network-metrics add-network-metrics-v2` also shows an empty result 👍

**Review by geofffranks — APPROVED** [review]
